src/Tribe/Aggregator/Migrate.php

```diff
@@ -0,0 +1,434 @@
+	}
+
+	/**
+	 * Checks if one or any iCal settings exists
+	 *
+	 * @param string|null $index If null will return a Object with all the legacy settings
+	 *
+	 * @return bool
+	 */
+	public function has_ical_setting( $index = null ) {
+		// Fetches the Settings, and remove the empty Ones
+		$values = array_filter( (array) $this->get_ical_setting( $index ), array( $this, 'is_empty' ) );
+
+		// if it's empty means we have empty legacy settings
+		return ! empty( $values );
+	}
+
+	/**
+	 * Checks if legacy Facebook settings were migrated
+	 *
+	 * @return bool
+	 */
+	public function is_ical_migrated() {
+		$records = Tribe__Events__Aggregator__Records::instance();
+
+		$args = array(
+			'post_status'    => Tribe__Events__Aggregator__Records::$status->schedule,
+			'posts_per_page' => 1,
+			'post_mime_type' => 'ea/ical',
+			'meta_query'     => array(
+				array(
+					'key'     => $records->prefix_meta( 'is_legacy' ),
+					'compare' => 'EXISTS',
+				),
+			),
+		);
+
+		return $records->query( $args )->have_posts();
+	}
+
+	/**
+	 * Method that Handles the AJAX converting of Legacy Facebook Settings
+	 * AJAX methods will not return anything, only print a JSON string
+	 *
+	 * @return void
+	 */
+	public function ajax_convert_facebook_settings() {
+		$response = (object) array(
+			'status' => false,
+			'text' => esc_html__( 'Error, a unknown bug happened and it was impossible to migrate the old Facebook Settings to Event Aggregator, try again later.', 'the-events-calendar' ),
+		);
+
+		$post_type = get_post_type_object( Tribe__Events__Main::POSTTYPE );
+
+		if ( empty( $post_type->cap->edit_posts ) || ! current_user_can( $post_type->cap->edit_posts ) ) {
+			$response->status = false;
+			$response->text = esc_html__( 'You do not have permission to migrate the old Facebook Settings to Event Aggregator', 'the-events-calendar' );
+
+			wp_send_json( $response );
+		}
+
+		if ( ! $this->has_facebook_setting() ) {
+			$response->status = false;
+			$response->text = esc_html__( 'It does not seem like there are any Legacy Facebook settings to be imported', 'the-events-calendar' );
+
+			wp_send_json( $response );
+		}
+
+		$settings = $this->get_facebook_setting();
+
+		$status = (object) array(
+			'error' => array(),
+			'success' => array(),
+		);
+
+		$origin = 'facebook';
+
+		foreach ( $settings->ids as $source ) {
+			$record = Tribe__Events__Aggregator__Records::instance()->get_by_origin( $origin );
+
+			/**
+			 * @todo Include the Deactivated logic
+			 */
+			$meta = array(
+				'origin'       => $origin,
+				'type'         => 'schedule',
+				'frequency'    => $this->convert_ical_frequency( $settings->frequency ),
+				'file'         => null,
+				'keywords'     => null,
+				'location'     => null,
+				'start'        => null,
+				'radius'       => null,
+				'source'       => $source,
+				'content_type' => null,
+				'is_legacy'    => true,
+				'import_id'    => null,
+				'post_status'  => $settings->post_status,
```

Comment:
Legacy iCal imports can have keywords, location, start, and radius. We should make sure those values are added here if they were set.